### PR TITLE
fail envoy.deprecated_features.v1_filter_json_config by default

### PR DESCRIPTION
**Pull request by mattklein123**
Risk Level: Low
Testing: Existing tests
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

**Comment by mattklein123** [comment]
Will fix up the failing test tomorrow. No need to review until then.

/wait

**Comment by mattklein123** [comment]
@alyssawilk this should be ready to review. cc @derekargueta after we cut the next release I think we can delete all of V1. :hocho:

File: source/common/runtime/runtime_features.cc [hunk 1]
```diff
@@ -41,18 +41,19 @@ constexpr const char* runtime_features[] = {
...
-    "envoy.deprecated_features.route.proto:enabled",
     "envoy.deprecated_features.fault.proto:type",
+    "envoy.deprecated_features.lds.proto:use_original_dst",
+    "envoy.deprecated_features.redis_proxy.proto:catch_all_cluster",
+    "envoy.deprecated_features.redis_proxy.proto:cluster",
+    "envoy.deprecated_features.route.proto:enabled",
     "envoy.deprecated_features.route.proto:runtime_key",
+    "envoy.deprecated_features.server_info.proto:max_obj_name_len",
+    "envoy.deprecated_features.server_info.proto:max_stats",
+    "envoy.deprecated_features.v1_filter_json_config",
```
Comment (alyssawilk) [inline source/common/runtime/runtime_features.cc]: Do we want to alphabetize or organize per release?

Comment (mattklein123) [reply source/common/runtime/runtime_features.cc]: I'm up for either, happy to put in comments per-release, or just alpha the whole thing. LMK what you prefer.

**Comment by mattklein123** [comment]
> I still want to know why the failure manifested as a failure to list the stats prefix rather than some conversion failure, but it's a non-blocking point of curiosity :-P

Unfortunate confusing error messages in proto-gen-validate. That's the first required field that wasn't set...

> You up for envoy-announcing with merge?

Yes on my list!

File: source/common/runtime/runtime_features.cc (hunk 1 above)
Comment (alyssawilk) [reply source/common/runtime/runtime_features.cc]: I think I'd prefer comments per-release, or changing the scrip to do alpha order so we don't have to do it manually each time.  Happy to go either way.

Comment (mattklein123) [reply source/common/runtime/runtime_features.cc]: OK I will fix.

**Comment by mattklein123** [comment]
This is failing on the coverage limit. I will figure something out.

**Comment by alyssawilk** [comment]
Didn't lizan merge a coverage fix last night?  I didn't catch up on the
details but would master merge help?

**Comment by mattklein123** [comment]
I'm trying to delete dead code to get coverage to pass, but this will take a while. Putting to waiting for now while I sort this out. 

/wait

**Comment by mattklein123** [comment]
@alyssawilk unfortunately converting all of these configs over has lost us a lot of coverage on the JSON config conversion code, which we technically cannot delete until we can fully undo the deprecated v1 configs (this primary change). I will see how close I get after this round of deletions, but I might need to drop the coverage to 96.8 to get this through in a reasonable amount of time. Then we can try to pick up more coverage elsewhere. cc @derekargueta

**Comment by alyssawilk** [comment]
If we need to lower coverage for the duration that's Ok, though it's a bummer we could theoretically accidentally break that code between flag flip and removal.

This is basically the situation I added the  "--runtime-feature-override-for-tests=" per test flag for, though I suspect it won't work because it's an sh test not a cc test.  Curses.  If it's not a pain we could tweak the hot restart runtime bootstrap to flip the flag back, just for coverage.  I'll leave it up to you if you think it's worth it.

**Comment by mattklein123** [comment]
I think what I'm going to do is add back a single HCM config test that exercises the old code path, which should fix coverage and we can delete it when we kill v1 completely. I will split out the other cleanups I've done into a dedicated PR. cc @derekargueta

**Comment by mattklein123** [comment]
@alyssawilk I think this will do the trick. I added an explicit test of the old v1 path to the HCM config test so this should get us all the same coverage that we had previously and do it in a place that is trivial to cleanup when everything gets deleted.

**Comment by mattklein123** [comment]
Failures are unrelated flakes and coverage looks good so going to merge.

**Pull request merged**